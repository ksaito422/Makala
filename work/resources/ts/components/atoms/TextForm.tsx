import React from 'react';
import { TextField } from '@material-ui/core';

type Props = Partial<{
  error: boolean;
  helperText: string;
  multiline: boolean;
  fullWidth: boolean;
  rowsMax: number;
  rows: number;
  defaultValue: string;
  value: string;
  margin: 'none' | 'dense' | 'normal';
  label: string;
  name: string;
  type: string;
  autoComplete: string;
  required: boolean;
  autoFocus: boolean;
  inputRef: any;
  disabled: boolean;
}>;

export const TextForm: React.FC<Props> = (props) => {
  return (
    <TextField
      color='secondary'
      variant='filled'
      error={props.error}
      helperText={props.helperText}
      multiline={props.multiline}
      fullWidth={props.fullWidth}
      rowsMax={props.rowsMax}
      rows={props.rows}
      defaultValue={props.defaultValue}
      value={props.value}
      margin={props.margin}
      label={props.label}
      name={props.name}
      type={props.type}
      autoComplete={props.autoComplete}
      required={props.required}
      autoFocus={props.autoFocus}
      inputRef={props.inputRef}
      disabled={props.disabled}
    />
  );
};
