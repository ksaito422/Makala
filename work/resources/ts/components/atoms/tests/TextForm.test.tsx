import React from 'react';
import { render } from '@testing-library/react';
import { TextForm } from '../TextForm';

describe('TextForm', () => {
  it('スナップショットテスト', () => {
    const props = {
      error: true,
      helperText: 'helperText',
      rowsMax: 4,
      rows: 2,
      defaultValue: 'defaultValue',
      label: 'label',
      name: 'name',
      autoComplete: 'autoComplete',
      required: true,
      autoFocus: true,
    };

    const { baseElement } = render(<TextForm {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
