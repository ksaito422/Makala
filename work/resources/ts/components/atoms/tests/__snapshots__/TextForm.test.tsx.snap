// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextForm スナップショットテスト 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled Mui-error Mui-error MuiFormLabel-filled Mui-focused Mui-focused Mui-required Mui-required"
        data-shrink="true"
      >
        label
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error Mui-error"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorSecondary MuiFilledInput-colorSecondary Mui-error Mui-error Mui-focused Mui-focused MuiInputBase-formControl"
      >
        <input
          aria-invalid="true"
          autocomplete="autoComplete"
          class="MuiInputBase-input MuiFilledInput-input"
          name="name"
          required=""
          rows="2"
          type="text"
          value="defaultValue"
        />
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error MuiFormHelperText-filled Mui-focused Mui-required"
      >
        helperText
      </p>
    </div>
  </div>
</body>
`;
